internal/controlplane/handlers_profile.go

```diff
@@ -964,3 +1071,76 @@ func deleteRuleStatusesForProfile(
 
 	return nil
 }
+
+func computeRuleName(rule *minderv1.Profile_Rule) string {
+	if rule.GetName() != "" {
+		return rule.GetName()
+	}
+
+	return rule.GetType()
+}
+
+func populateRuleNames(profile *minderv1.Profile) {
+	_ = engine.TraverseAllRulesForPipeline(profile, func(r *minderv1.Profile_Rule) error {
+		r.Name = computeRuleName(r)
+		return nil
+	},
+	)
+}
+
+func validateRuleNameAndTypeInProfile(profile *minderv1.Profile) error {
+	for ent, entRules := range map[minderv1.Entity][]*minderv1.Profile_Rule{
+		minderv1.Entity_ENTITY_REPOSITORIES:       profile.GetRepository(),
+		minderv1.Entity_ENTITY_ARTIFACTS:          profile.GetArtifact(),
+		minderv1.Entity_ENTITY_BUILD_ENVIRONMENTS: profile.GetBuildEnvironment(),
+		minderv1.Entity_ENTITY_PULL_REQUESTS:      profile.GetPullRequest(),
+	} {
+		if err := validateRuleNameAndType(ent, entRules); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateRuleNameAndType(entity minderv1.Entity, rules []*minderv1.Profile_Rule) error {
+	ruleNameToType := make(map[string]string)
+	emptyNameTypesSet := sets.New[string]()
+	for _, rule := range rules {
+		ruleName := rule.GetName()
+		ruleType := rule.GetType()
+
+		if ruleName == ruleType {
+			return &engine.RuleValidationError{
+				Err:      fmt.Sprintf("cannot have rules with same name and type in entity '%s'", entity.ToString()),
+				RuleType: ruleType,
+			}
+		}
```

Comment:
This is there to invalidate following types of profiles:
```yaml
repository:
  - type: dependabot_configured
    name: dependabot_configured
    def:
      package_ecosystem: gomod
  - type: dependabot_configured
    def:
      package_ecosystem: npm
```
And also as a general rule to make sure users have proper names in profiles.

Reply:
You're also invalidating the following profile, though:

```
repository:
  - type: dependabot_configured
    name: dependabot_configured
    def:
      package_ecosystem: gomod
```

(Which is what we'll be storing after this PR.  But actually writing the `name` field will get you an error...)

Reply:
This was intentional to prevent non required fields on the users' side. But yeah, this should not be an error, user can be verbose if they want to be.